wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -326,9 +326,9 @@ public void onChanged(Resource<Boolean> status) {
         AppDatabase.getAppDatabase().blockchainIdentityDataDao().load().observe(this, new Observer<BlockchainIdentityData>() {
             @Override
             public void onChanged(BlockchainIdentityData blockchainIdentityData) {
-                if(blockchainIdentityData != null) {
+                if (blockchainIdentityData != null) {
                     BlockchainIdentity.RegistrationStatus status = blockchainIdentityData.getRegistrationStatus();
-                    hasIdentity = status != null ? status == BlockchainIdentity.RegistrationStatus.REGISTERED : false;
+                    hasIdentity = (status == BlockchainIdentity.RegistrationStatus.REGISTERED);
```

Comment:
I think I had a check for `status == null` because it could be null, perhaps this problem is addressed elsewhere.

Reply:
yea, basically it doesn't matter since `==` operator doesn't throw NPE